Add chapter to migration guide on how to migrate to Volto (#1254)
plone/documentation: docs/backend/upgrading/upgrade-to-volto.md

```{todo}
Is this "upgrading" or "migrating"? It seems to be a migration more than an upgrade. Please clarify.

Technically it is a migration since the version of Plone does not get changed and it is no upgrade-step.
It could be called a upgrade though because Volto is the new default frontend.

> Not 100% sure...

Reviewer: I think that "upgrade" is less correct than "migration". In the instructions, we "Migrate RichText fields to Volto blocks" and the form's URI contains "migrate_to_volto". Let's use migrate instead of upgrade everywhere in this document, including its file name.

Admittedly it is a little bit weird because it is located within the "upgrading" directory, but I'm OK with a URI of `/backend/upgrading/migrate-to-volto.html`.

Author: Ok, I'm fine with using the term Migration for this :)

```

# Upgrading to Volto

Plone 6 comes with a new default frontend, called Volto.
Volto is written in React and uses `plone.restapi` to communicate with the backend.

> When creating a new site in Plone 6 you have the option to choose between creating a Volto site and a Classic Plone site.

Reviewer: Out of curiosity, what is the actual terminology used in the UI?

In my mind, one creates a Plone site with a frontend of either Volto or Classic UI. I dislike the phrase of "creating a Volto site" or "creating a Classic Plone site" as it implies that they are two totally different sites instead of just UIs. Both options create a "Plone site". Only the UI is the primary difference.

```suggestion
When creating a new Plone 6 site, you may choose between Volto and Classic UI for the frontend.
```

Author: I actually is "Create a new Plone site" and "Create Classic Plone site":

<img width="571" alt="Bildschirmfoto 2022-05-17 um 19 06 53" src="https://user-images.githubusercontent.com/453208/168870925-5f93b9af-87c4-424a-868a-b2f871f9aa77.png">

I'll update accordingly

Reviewer: Where would be a good place to discuss this UI? I think it could be improved. I approach this with fresh eyes, and it is... confusing.

For this PR, aligning with the UI is the way forward.

Reviewer: Found it, and not easily.

https://github.com/plone/Products.CMFPlone/commit/cd57b6e82fca76a35caac018a18211e7a7dddc39

> This is because there are some non-trivial differences between the configuration of a Classic Plone and a Volto site.

Reviewer: ```suggestion
This choice is presented because there are some non-trivial differences between their configurations.
```

> This document discusses these differences and tells admins and developers how to modify their existing Plone site to be compatible with Volto.

Reviewer: ```suggestion
This document discusses these differences.
It also informs administrators and developers of how to migrate their existing Plone 6 site with Classic UI for its frontend to instead become compatible with Volto for its frontend.
```

> Pre-requisites: The site needs to be upgraded to Plone 6 before

Reviewer: ````suggestion
```important
As a pre-requisite, your Plone site must be [upgraded to Plone 6](v60) before migrating to Volto for the frontend.
```
````

> Plone provides a form `/@@migrate_to_volto` that allows you to run all the required changes to your existing site to make it compatible with Volto.

Reviewer: ```suggestion
Plone provides a form `/@@migrate_to_volto` that allows you to run all the required changes to your existing Plone 6 site to make it compatible with Volto.
```

> You can access this form in the browser when you are logged-in as an administrator by opening `http://localhost:8080/Plone/@@migrate_to_volto` (where `localhost` is your server, `8080` the port that Plone is running on and `Plone` is the name of the Plone instance).

Reviewer: ```suggestion
You can access this form in the browser when you are logged in as an administrator by opening `http://localhost:8080/Plone/@@migrate_to_volto` (where `localhost` is your server, `8080` is the port on which Plone is running, and `Plone` is the name of the Plone instance).
```

> After upgrading an existing site to Plone 6 (see {doc}`v60`) there will also appear a message **You can prepare your site for Volto, the default frontend of Plone 6!** with a link to that form.

Reviewer: ```suggestion
Additionally, after upgrading an existing site to Plone 6 (see {doc}`v60`) a message will appear, **You can prepare your site for Volto, the default frontend of Plone 6!**, with a link to that form.
```

```{warning}
Test all migrations thoroughly before applying them on a production environment!

> A site that is make compatible with Volto will be accessible with Plone Classic UI, but it will behave differently.

Reviewer: ```suggestion
A site that is made compatible with Volto will be accessible with Plone Classic UI, but it will behave differently.
```

> For example, editors can only effectively work with the content using Volto because html is no longer editable in the TinyMCE editor.

Reviewer: ```suggestion
For example, editors can only effectively work with the content using Volto because HTML is no longer editable in the TinyMCE editor used only in Classic UI.
```

```

```{todo}
`{py:mod}` does not link to the Python modules (see https://deploy-preview-1254--6-dev-docs-plone-org.netlify.app/backend/upgrading/upgrade_to_volto.html).

We could either use inline literals as shown in the suggestion, or we could link to their Glossary entry (or create one), GitHub repo, PyPI project, or relevant chapter in the docs. Which do you think is best?

> I like `{py:mod}` because it is semantic and renders like a inline-literal. We don't require link in these cases.

Reviewer: Thanks for the explanation, and that makes sense to me now. Please delete the todo discussion for this one.

    The criteria of the collection are uses to configure a listing block in that page.

It is recommended to use the default settings, but you can choose to skip some of the migration steps in the form.

```{note}

> If you are migrating a existing site to Plone 6 using [{py:mod}`collective.exportimport`](https://pypi.org/project/collective.exportimport) and want to use Volto in the new site

Reviewer: ```suggestion
If you are migrating an existing site to Plone 6 using [{py:mod}`collective.exportimport`](https://pypi.org/project/collective.exportimport) and want to use Volto in the new site, then you do not need to use the form `@@migrate_to_volto`.
```

> you do not need to use the form `@@migrate_to_volto`.

Reviewer: Merge with previous line.

```suggestion
```

> All the changes documented above can be done efficiently during export and import and are discussed [here](https://github.com/collective/collective.exportimport/issues/133).

Reviewer: ```suggestion
All the changes documented above can be done efficiently during export and import.
[Read details](https://github.com/collective/collective.exportimport/issues/133).
```
